BUILDER_DEV.md

```diff
@@ -1,225 +1,109 @@
-export HAB_BLDR_URL="http://localhost:9636"
-export HAB_ORIGIN="core"
-```
+1. [Setup a GitHub application](https://github.com/settings/apps/new) for your GitHub organization.
+1. Set the value of `Homepage URL` to `http://${APP_HOSTNAME}`
+1. Set the value of `User authorization callback URL` to `http://${APP_HOSTNAME}/` (The trailing `/` is *important*)
+1. Set the value of `Webhook URL` to `http://${APP_HOSTNAME}/`
+1. Set everything to read only (this is only used for your org so it's safe)
+1. Save and download the pem key
+1. Copy the pem key to `${HABITAT_SRC_ROOT}/.secrets/builder-github-app.pem`
+1. Record the the client-id, client-secret, app_id and webhook-secret. These will be used for the `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET`, `GITHUP_APP_ID` and `GITHUB_WEBHOOK_SECRET` build variables (seen below).
 
-* symlink /src to path/to/your/habitat/repo
+### Create app env file
 
+1. Create `${HABITAT_SRC_ROOT}/.secrets/habitat-dev`
+1. Add
 ```
-$ sudo ln -s path/to/your/habitat/repo /src
+  APP_HOSTNAME=localhost
+  GITHUB_API_URL=https://api.github.com
+  GITHUB_WEB_URL=https://github.com
+  GITHUB_CLIENT_ID=${GITHUB_CLIENT_ID}
+  GITHUB_CLIENT_SECRET=${GITHUB_CLIENT_SECRET}
+  GITHUB_WEBHOOK_SECRET=${GITHUB_WEBHOOK_SECRET}
+  GITHUB_APP_ID=${GITHUP_APP_ID}
 ```
+1. Save and close
 
-* Cd into your Habitat repo
+### Studio requirements
 
-```
-$ cd /src
-```
+1. Ensure you have run `hab setup` in your environment that will be executing the studio
 
-* Run the provision script
+### Starting the services
 
-```
-$ ./support/linux/provision.sh
-```
+From either your VM or Mac:
 
-* Source cargo
-
-```
-$ source $HOME/.cargo/env
-```
+* `cd <habitat source path>`
+* `direnv allow`
+* `hab studio enter`
+* `start-builder`
 
-* Run the build-srv makefile (NOTE - if you receive an error about cargo not being found, make sure you have removed the "secure_path" section of /etc/sudoers)
-
-```
-$ make build-srv
-```
-
-* Run this command (it may take awhile) to start all the Builder services
-
-```
-$ sudo -E make bldr-run-no-build
-```
+### UI Setup
 
-* Wait for all services to start - they are started when you see worker heartbeat debug entries in the log. If services do not start because some were already running from a previous attempt, run `$ make bldr-kill`
+If you are not doing UI development you just need to navigate to `${APP_HOSTNAME}/#/sign-in`
 
-### UI Setup
+If you are developing the UI:
 
 * Follow the instructions in the [Web UI README](https://github.com/habitat-sh/habitat/blob/master/components/builder-web/README.md) to get the Web UI running locally.
 * Open up a browser and navigate to http://localhost:3000/#/pkgs - you should see the Habitat UI running locally.
 
-### Creating an Origin
+## Helper functions
 
-### Option A: Create an origin using the web UI
+* `start-builder` - Run the builder cluster
+* `origin` - Create the core origin
+* `project` - Create a project (you can also configure this in the web UI)
```

Comment:
Not your issue, but the `project` function is broken:
```
project () 
{ 
    curl http://localhost:9636/v1/projects -q -d '{ "origin": "core", "plan_path": "nginx/plan.sh", "github": { "organization": "habitat-sh", "repo": "core-plans" } }';
    -H "Authorization:Bearer:${HAB_AUTH_TOKEN}"
}
```

1. The `-H` arg should be part of the `curl` command (and might as well throw in a `--fail` there too)
2. It's weird that this `project` command specifically creates a `nginx` project
3. Even fixing the auth, I get a failure:

```
[53][default:/src:22]# curl -v --fail http://localhost:9636/v1/projects -q -d '{ "origin": "core", "plan_path": "nginx/plan.sh", "github": { "organization": "habitat-sh", "repo": "core-plans" } }' -H "Authorization:Bearer:${HAB_AUTH_TOKEN}"
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 9636 (#0)
> POST /v1/projects HTTP/1.1
> Host: localhost:9636
> User-Agent: curl/7.54.1
> Accept: */*
> Authorization:Bearer:REDACTED
> Content-Length: 116
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 116 out of 116 bytes
* The requested URL returned error: 422 Unprocessable Entity
* stopped the pause stream!
* Closing connection 0
curl: (22) The requested URL returned error: 422 Unprocessable Entity
```

Reply:
I thought at first this was because the `core` origin for my local builder instance had no keys associated, but I generated and uploaded them (unintentionally rotating the private key for production builder in the process 😬), but I still get the same error.

Reply:
Fixed the missing backslash